src/sentry/grouping/fingerprinting/__init__.py

```diff
@@ -470,17 +381,112 @@ def text(self):
         ).rstrip()
 
 
-class FingerprintingVisitor(NodeVisitor):
+class FingerprintingRules:
+    def __init__(
+        self,
+        rules: Sequence[Rule],
+        changelog: Sequence[Any] = None,
+        version: int = None,
+        bases: Sequence[Any] | None = None,
+    ) -> None:
+        if version is None:
+            version = VERSION
+        self.version = version
+        self.rules = rules
+        self.changelog = changelog
+        self.bases = bases or []
+
+    def iter_rules(self, include_builtin: bool = True) -> Generator[Rule, Any, Any]:
```

Comment:
Is this return type right? It looks so odd to me.

Reply:
it should be `Generator[Rule, None, None]` sine there is no `Send` or `Return` type -- https://youtu.be/DTegfCNAXoM